src/sagas/index.js

```diff
@@ -258,21 +255,21 @@ function* fetchAndPutPools (action) {
 
 export function* fetchSinglePool (action) {
   const pool = yield callExternalAction(Api.getPool, action, true)
+  if (pool?.error) {
+    return { poolId: action.payload.poolId, error: pool.error }
+  }
 
-  const internalPool = pool.id ? Transforms.Pool.toInternal({ pool }) : null
-  const error = pool.error ? pool.error.status ?? 'fetch-error' : null
+  const internalPool = Transforms.Pool.toInternal({ pool })
 
-  return { poolId: action.payload.poolId, internalPool, error }
+  return { poolId: action.payload.poolId, internalPool }
 }
 
 function* fetchAndPutSinglePool (action) {
   const { internalPool, error } = yield fetchSinglePool(action)
 
-  if (error) {
-    if (error === 404) {
-      yield put(updateVms({ removePoolIds: [action.payload.poolId] }))
-    }
-  } else {
+  if (error?.status === 404) {
+    yield put(updateVms({ removePoolIds: [action.payload.poolId] }))
+  } else if (!error && internalPool) {
```

Comment:
I prefer the old if structure to the new one ... `if (error) { do errors } else { do success }`.